[Feat] Add plain text component (#1061)
mckinsey/vizro: vizro-core/docs/pages/user-guides/markdown.md

# How to use markdown

This guide shows you how to use markdown to visualize your data in the dashboard.

> The [`Markdown`][vizro.models.Markdown] is a flexible and extensible component that enables customization via markdown text. Refer to any online guide for [basic markdown usage](https://markdown-guide.readthedocs.io/en/latest/). It is based on the underlying Dash component [`dcc.Markdown`](https://dash.plotly.com/dash-core-components/markdown/).

Reviewer: ```suggestion
The [`Markdown` component][vizro.models.Markdown] is flexible and extensible and is for use in customizing markdown text. To find out more about basic Markdown syntax, refer to [this or any other [online guide](https://markdown-guide.readthedocs.io/en/latest/). 

The `Markdown` component is based on the underlying Dash component [`dcc.Markdown`](https://dash.plotly.com/dash-core-components/markdown/).
```

> You can add a [`Markdown`][vizro.models.Markdown] to your dashboard by inserting the [`Markdown`][vizro.models.Markdown] into the `components` argument of the [`Page`][vizro.models.Page] or of the [`Container`][vizro.models.Container] model.

Reviewer: ```suggestion
You can add a [`Markdown`][vizro.models.Markdown] component to your dashboard by inserting [`Markdown`][vizro.models.Markdown] into the `components` argument of the [`Page`][vizro.models.Page] or the [`Container`][vizro.models.Container] model.
```

!!! example "Markdown"
    === "app.py"
        ```{.python pycafe-link}
        import vizro.models as vm
        from vizro import Vizro

        page = vm.Page(
            title="Markdown",
            components=[
                vm.Markdown(
                    text="""
                        ### Markdown Title
                        Commodi repudiandae consequuntur voluptatum.
                    """,
                ),
            ],
        )

        dashboard = vm.Dashboard(pages=[page])

        Vizro().build(dashboard).run()
        ```

    === "app.yaml"
        ```yaml
        # Still requires a .py to add data to the data manager and parse YAML configuration
        # See from_yaml example
        pages:
          - components:
              - text: |
                  Commodi repudiandae consequuntur voluptatum.
                title: Markdown Title
                type: markdown
            title: Markdown
        ```

    === "Result"
        [![Markdown]][markdown]

> ### Customize Markdown text

Reviewer: ```suggestion
## Customize Markdown text
```

> The [`Markdown`][vizro.models.Markdown] uses the `dcc.Markdown` component from Dash as its underlying text component. For more details on customizing the markdown text, refer to the [`dcc.Markdown` component documentation](https://dash.plotly.com/dash-core-components/markdown). Based on examples from Dash, the [`Markdown`][vizro.models.Markdown] model supports the following:

Reviewer: ```suggestion
The [`Markdown`][vizro.models.Markdown] component uses the `dcc.Markdown` component from Dash as its underlying text component. For more details on customizing markdown text, refer to the [`dcc.Markdown` component documentation](https://dash.plotly.com/dash-core-components/markdown). As it is based on examples from Dash, the [`Markdown`][vizro.models.Markdown] model supports the following:
```

> - Headers

Reviewer: Images? Links?

!!! example "Markdown with custom text"
    === "app.py"
        ```{.python pycafe-link}
        import vizro.models as vm
        from vizro import Vizro

        page = vm.Page(
            title="Customizing Text",
            components=[
                vm.Markdown(
                    text="""
                        # Header level 1 <h1>

                        ## Header level 2 <h2>

                        ### Header level 3 <h3>

                        #### Header level 4 <h4>
                    """,
                ),
            ],
        )

        dashboard = vm.Dashboard(pages=[page])

        Vizro().build(dashboard).run()
        ```

    === "app.yaml"
        ```yaml
        # Still requires a .py to add data to the data manager and parse YAML configuration
        # See from_yaml example
        pages:
          - components:
              - text: |
                  # Header level 1 <h1>

                  ## Header level 2 <h2>

                  ### Header level 3 <h3>

                  #### Header level 4 <h4>
                type: markdown
            title: Customizing Text
        ```

    === "Result"
        [![MarkdownText]][markdowntext]

> ### The `extra` argument

Reviewer: ```suggestion
## The `extra` argument
```

> The `Markdown` is based on the underlying Dash component [`dcc.Markdown`](https://dash.plotly.com/dash-core-components/markdown/). Using the `extra` argument you can pass additional arguments to `dcc.Markdown` in order to alter it beyond the chosen defaults.

Reviewer: ```suggestion
Use the `extra` argument you can pass additional arguments to `dcc.Markdown` to change the defaults.
```

# How to use markdown

> This guide shows you how to use markdown to visualize your data in the dashboard.

Reviewer 2: ```suggestion
This guide shows you how to use markdown to display plain text without any additional styling like borders or background in the dashboard.
```

> The [`Markdown` component][vizro.models.Markdown] is flexible and extensible and is for use in customizing markdown text. To find out more about basic Markdown syntax, refer to [this or any other online guide](https://markdown-guide.readthedocs.io/en/latest/).

Reviewer 2: ```suggestion
The [`Markdown` component][vizro.models.Markdown] is used for displaying plain text. It is flexible and extensible due to its underlying usage of Markdown syntax. To find out more about basic Markdown syntax, refer to [this or any other online guide](https://markdown-guide.readthedocs.io/en/latest/).
```

!!! example "Markdown"
    === "app.py"
        ```{.python pycafe-link}
        import vizro.models as vm
        from vizro import Vizro

        page = vm.Page(
            title="Markdown",
            components=[
                vm.Markdown(
                    text="""
                        ### Markdown Title
                        Commodi repudiandae consequuntur voluptatum.
                    """,
                ),
            ],
        )

        dashboard = vm.Dashboard(pages=[page])

        Vizro().build(dashboard).run()
        ```

    === "app.yaml"
        ```yaml
        # Still requires a .py to add data to the data manager and parse YAML configuration
        # See from_yaml example
        pages:
          - components:
              - text: |
                  Commodi repudiandae consequuntur voluptatum.
                title: Markdown Title
                type: markdown
            title: Markdown
        ```

    === "Result"
        [![Markdown]][markdown]

## Customize Markdown text

> The [`Markdown`][vizro.models.Markdown] component uses the `dcc.Markdown` component from Dash as its underlying text component. For more details on customizing markdown text, refer to the [`dcc.Markdown` component documentation](https://dash.plotly.com/dash-core-components/markdown). As it is based on examples from Dash, the [`Markdown`][vizro.models.Markdown] model supports the following:

Reviewer 2: ```suggestion
The [`Markdown`][vizro.models.Markdown] component uses the `dcc.Markdown` component from Dash as its underlying text component. For more details on customizing markdown text, refer to the [`dcc.Markdown` component documentation](https://dash.plotly.com/dash-core-components/markdown). It supports (among other things) the following features:
```

    === "app.py"
        ```{.python pycafe-link}
        import vizro.models as vm
        from vizro import Vizro

        page = vm.Page(
            title="Customizing Text",
            components=[
                vm.Markdown(
                    text="""
                        # Header level 1 <h1>

                        ## Header level 2 <h2>

                        ### Header level 3 <h3>

                        #### Header level 4 <h4>
                    """,
                ),
            ],
        )

        dashboard = vm.Dashboard(pages=[page])

        Vizro().build(dashboard).run()
        ```

    === "app.yaml"
        ```yaml
        # Still requires a .py to add data to the data manager and parse YAML configuration
        # See from_yaml example
        pages:
          - components:
              - text: |
                  # Header level 1 <h1>

                  ## Header level 2 <h2>

                  ### Header level 3 <h3>

                  #### Header level 4 <h4>
                type: markdown
            title: Customizing Text
        ```

    === "Result"
        [![MarkdownText]][markdowntext]

## The `extra` argument

> Use the `extra` argument to pass additional arguments to `dcc.Markdown` to change the defaults.

Reviewer 2: can we make it more like the other components: The Button is based on the underlying Dash component [dbc.Button](https://dash-bootstrap-components.opensource.faculty.ai/docs/components/button/). Using the extra argument you can pass additional arguments to dbc.Button in order to alter it beyond the chosen defaults.

Ie can we also add a link to make it very clear?

!!! note
    Using `extra` is a quick and flexible way to alter a component beyond what Vizro offers. However, [it is not a part of the official Vizro schema](../explanation/schema.md#what-is-the-vizro-json-schema) and the underlying implementation details may change. If you want to guarantee that your apps keep running, we recommend that you pin your Vizro version.

An example use would be to set `Markdown` `mathjax=True` (defaults to `False`) to display the mathematical equations. For this, you can use `extra={"mathjax": True}`.

!!! example "Markdown with extra argument"
    === "app.py"
        ```{.python pycafe-link}
        import vizro.models as vm
        from vizro import Vizro

        page = vm.Page(
            title="Customizing Markdown",
            components=[
                vm.Markdown(
                    text="""
                      This example uses the block delimiter:
                      $$

>                       \\int_0^\\infty e^{-x^2} dx = \\frac{\\sqrt{\\pi}}{2}

Reviewer 2: 😍
